modules/REST/V1/Options.php

```diff
@@ -0,0 +1,218 @@
+				);
+				continue;
+			}
+
+			$result = $this->options->set( $option_name, $request[ $option_name ] );
+
+			if ( $result->has_errors() ) {
+				$errors->merge_from( $result );
+			} else {
+				$response[ $option_name ] = $this->options->get( $option_name );
+			}
+		}
+
+		if ( $errors->has_errors() ) {
+			return $this->add_status_to_error( $errors );
+		}
+
+		return $this->prepare_item_for_response( $response, $request );
+	}
+
+	/**
+	 * Checks if a given request has access to get the options.
+	 *
+	 * @since 3.7
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return true|WP_Error True if the request has read access, WP_Error object otherwise.
+	 *
+	 * @phpstan-template T of array
+	 * @phpstan-param WP_REST_Request<T> $request
+	 */
+	public function get_items_permissions_check( $request ) { // phpcs:ignore VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable
+		if ( ! $this->check_update_permission() ) {
+			return new WP_Error(
+				'rest_forbidden_context',
+				__( 'Sorry, you are not allowed to edit options.', 'polylang' ),
+				array( 'status' => rest_authorization_required_code() )
+			);
+		}
+		return true;
+	}
+
+	/**
+	 * Checks if a given request has access to update the options.
+	 *
+	 * @since 3.7
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return true|WP_Error True if the request has access to update the option, WP_Error object otherwise.
+	 *
+	 * @phpstan-template T of array
+	 * @phpstan-param WP_REST_Request<T> $request
+	 */
+	public function update_items_permissions_check( $request ) { // phpcs:ignore VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable
+		return $this->get_items_permissions_check( $request );
+	}
+
+	/**
+	 * Prepares the option value for the REST response.
+	 *
+	 * @since 3.7
+	 *
+	 * @param array           $item    Option values.
+	 * @param WP_REST_Request $request Request object.
+	 * @return WP_REST_Response Response object.
+	 *
+	 * @phpstan-template T of array
+	 * @phpstan-param array<non-falsy-string, mixed> $item
+	 * @phpstan-param WP_REST_Request<T> $request
+	 */
+	public function prepare_item_for_response( $item, $request ) {
+		$fields   = $this->get_fields_for_response( $request );
+		$response = array();
+
+		foreach ( $item as $option => $value ) {
+			if ( rest_is_field_included( $option, $fields ) ) {
+				$response[ $option ] = $value;
+			}
+		}
+
+		/** @var WP_REST_Response */
+		return rest_ensure_response( $response );
+	}
+
+	/**
+	 * Retrieves the options' schema, conforming to JSON Schema.
+	 *
+	 * @since 3.7
+	 *
+	 * @return array Item schema data.
+	 */
+	public function get_item_schema(): array {
+		return $this->add_additional_fields_schema( $this->options->get_schema() );
+	}
+
+	/**
+	 * Tells if options can be edited.
+	 *
+	 * @since 3.7
+	 *
+	 * @return bool
+	 */
+	protected function check_update_permission(): bool {
```

Comment:
This method can be removed if I read correctly.
It's used only in `get_items_permissions_check()` and thus the code can be put in it directly.

Reply:
Right, I thought it was in `WP_REST_Controller` and should be implemented. Done in [73c7d58](https://github.com/polylang/polylang/pull/1569/commits/73c7d58bd3b984272a06520d2d9d4bf02dc9ad68).